pkg/admission/customresourcevalidation/securitycontextconstraints/validate_scc.go

```diff
@@ -0,0 +1,83 @@
+package securitycontextconstraints
+
+import (
+	"fmt"
+	"io"
+
+	"k8s.io/apimachinery/pkg/api/validation"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/apiserver/pkg/admission"
+
+	securityv1 "github.com/openshift/api/security/v1"
+
+	"github.com/openshift/origin/pkg/admission/customresourcevalidation"
+	securityvalidation "github.com/openshift/origin/pkg/admission/customresourcevalidation/security/validation"
+)
+
+const PluginName = "security.openshift.io/ValidateSecurityContextConstraint"
+
+func Register(plugins *admission.Plugins) {
+	plugins.Register(PluginName, func(config io.Reader) (admission.Interface, error) {
+		return customresourcevalidation.NewValidator(
+			map[schema.GroupResource]bool{
+				{Group: securityv1.GroupName, Resource: "securitycontextconstraints"}: true,
+			},
+			map[schema.GroupVersionKind]customresourcevalidation.ObjectValidator{
+				securityv1.GroupVersion.WithKind("SecurityContextConstraint"): securityContextConstraintV1{},
+			})
+	})
+}
+
+func toSecurityContextConstraint(uncastObj runtime.Object) (*securityv1.SecurityContextConstraint, field.ErrorList) {
+	if uncastObj == nil {
+		return nil, nil
+	}
+
+	allErrs := field.ErrorList{}
+
+	obj, ok := uncastObj.(*securityv1.SecurityContextConstraint)
+	if !ok {
+		return nil, append(allErrs,
+			field.NotSupported(field.NewPath("kind"), fmt.Sprintf("%T", uncastObj), []string{"SecurityContextConstraint"}),
+			field.NotSupported(field.NewPath("apiVersion"), fmt.Sprintf("%T", uncastObj), []string{securityv1.GroupVersion.String()}))
+	}
+
+	return obj, nil
+}
+
+type securityContextConstraintV1 struct {
+}
+
+func (securityContextConstraintV1) ValidateCreate(obj runtime.Object) field.ErrorList {
+	securityContextConstraintObj, errs := toSecurityContextConstraint(obj)
+	if len(errs) > 0 {
+		return errs
+	}
+
+	errs = append(errs, validation.ValidateObjectMeta(&securityContextConstraintObj.ObjectMeta, false, validation.NameIsDNSSubdomain, field.NewPath("metadata"))...)
+	errs = append(errs, quotavalidation.ValidateSecurityContextConstraint(securityContextConstraintObj)...)
```

Comment:
Pretty sure you did not mean `quotavalidation`. It's weird this compiled though (= image test is green) :thinking:

Reply:
wth :woman_shrugging:  yea, fixing